collector/fixtures/e2e-output.txt

```diff
@@ -785,43 +788,82 @@ node_md_blocks_synced{device="md10"} 3.14159265e+08
 node_md_blocks_synced{device="md11"} 4.190208e+06
 node_md_blocks_synced{device="md12"} 3.886394368e+09
 node_md_blocks_synced{device="md127"} 3.12319552e+08
+node_md_blocks_synced{device="md13"} 4.88383936e+08
+node_md_blocks_synced{device="md20"} 9.76751616e+08
 node_md_blocks_synced{device="md219"} 7932
 node_md_blocks_synced{device="md3"} 5.853468288e+09
 node_md_blocks_synced{device="md4"} 4.883648e+06
+node_md_blocks_synced{device="md50"} 8.790400512e+09
 node_md_blocks_synced{device="md6"} 1.6775552e+07
 node_md_blocks_synced{device="md7"} 7.813735424e+09
 node_md_blocks_synced{device="md8"} 1.6775552e+07
 node_md_blocks_synced{device="md9"} 523968
-# HELP node_md_disks Total number of disks of device.
+# HELP node_md_disks Number of disks of device. (state="any" equals to the number of disks in the array)
```

Comment:
I thought @bs-github was saying that there are scenarios in which only the information about the total size, but not about the exact breakdown in active/failed/missing devices are available. Though, after reading a lot more about mdstat, I agree that it should always be possible to read these information from the status line.

> But there could be more disks than that number, i.e. the spare disks.
So the sum would be bigger than total in that case.

To get the "total" number without spares, you'd write:

    sum without(state) (node_md_disks{state!="spare"})

I'll leave it to @SuperQ and @brian-brazil whether or not we should have a `node_md_desired_disks` metric.

> The number of missing disks is not reported explicitly and sometimes not at all.

Can you give examples? Reading about mdstat, it seems missing devices can be found by counting the number of devices in the device line and comparing that to the number of total devices (examples 2 and 6 in https://raid.wiki.kernel.org/index.php/Mdstat).